internal/venice-client-common/src/main/java/com/linkedin/venice/compression/GzipPool.java

```diff
@@ -0,0 +1,43 @@
+package com.linkedin.venice.compression;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+
+abstract class GzipPool {
+  private static final ThreadLocal<SettableOutputStream> streams =
+      ThreadLocal.withInitial(() -> new SettableOutputStream());
+
+  private static final ThreadLocal<ReusableGzipOutputStream> zipStreams =
+      ThreadLocal.withInitial(() -> new ReusableGzipOutputStream(streams.get()));
```

Comment:
Since these are always used together, could we group them into a static `ReusableObject` class, and have a single thread-local? Also, should we use a `ClosableThreadLocal` instead?

Reply:
Sure will club those two together. You mean use https://lucene.apache.org/core/3_0_3/api/all/org/apache/lucene/util/CloseableThreadLocal.html this ?

Reply:
We have our own implementation of `CloseableThreadLocal`, which Nisarg added recently.